### PR TITLE
Added sorting layers control

### PR DESCRIPTION
**Pull request by reesvarney**
This should allow anybody using this to apply the shadow caster to certain sorting layers rather than always going to all, personally I was using the sorting layers to apply shadows differently to objects and this wasn't possible - creating very harsh shadows.

I'm new to C# and unity so it may not be the most efficient however it seems operational.

### CODE REVIEW

**Comment by reesvarney** [comment]
I have just noticed that whilst it seems to retain its value and is functional, it resets the appearance on the dropdown back to nothing when previewing the game. I assume this is something to do with serialisation and I'll take a look into it.

**Comment by reesvarney** [comment]
Has been fixed by moving the calculation of layers into the runtime script. Only issue that might occur now would be the script not updating to match changes to the sorting layers.

**Comment by BrianCraig** [comment]
@reesvarney Hi! Sorry i didn't noticed the PR! I'll check in between 24hrs!

File: Editor/TilemapShadowCaster2DEditor.cs [hunk 1]
```diff
@@ -1,22 +1,39 @@
...
+using System.Collections.Generic;
 
 namespace TilemapShadowCaster.Editor
 {
     [CustomEditor(typeof(TilemapShadowCaster2D))]
     public class TilemapShadowCaster2DEditor : UnityEditor.Editor
     {
         private SerializedProperty m_selfShadowsProperty;
+        private SerializedProperty m_ApplyToSortingLayersProperty;
+        private static int flags;
```
Comment (BrianCraig) [inline Editor/TilemapShadowCaster2DEditor.cs]: flags, options and values are never used (here and OnEnable)

**Comment by BrianCraig** [comment]
@reesvarney Cool! Thanks for contributing!
Im going to add a readme to give more info and reference the [New example](https://github.com/BrianCraig/Tilemap-Shadow-Caster-Example), the code looks really good, don't mind i'm new into c# unity Editor coding also.

I added a non-breaking review, if you don't have time i'll accept the PR and change it after.

Brian

**Review by BrianCraig — CHANGES_REQUESTED** [review]

File: Editor/TilemapShadowCaster2DEditor.cs (hunk 1 above)
Comment (reesvarney) [reply Editor/TilemapShadowCaster2DEditor.cs]: Ah, I forgot to remove flags/ values as I was previously calculating the bitmask from the editor script. However, I think `options` needs to stay to create the maskfield in `OnInspectorGUI`:
```csharp
 int newVal = EditorGUILayout.MaskField("Apply to sorting layers", m_ApplyToSortingLayersProperty.intValue, options)
 ```

**Review by BrianCraig — APPROVED** [review]

**Comment by BrianCraig** [comment]
Cool, Thanks!

**Pull request merged**